### ISSUE TITLE
Procedural macro triggers `clippy::unreachable`

### ISSUE THREAD

**Issue by c410-f3r** [tokio-rs/tracing#2422]
```rust
#![deny(clippy::unreachable)]

#[tracing::instrument]
async fn _foo() {}
```

As seen above, the internal `instrument` implementation triggers `clippy::unreachable` in downstream projects. To avoid such behaviour, please consider tweaking the codebase or `#[allow(clippy::unreachable)]` where applicable.

**Comment by avandesa-fw** [comment]
HI, I found the same issue. Looks like it's fixed by #2356 but hasn't been released yet.

**Comment by c410-f3r** [comment]
Thank you
